src/lib/app.js

```diff
@@ -116,6 +116,8 @@ class App {
 
     router.get('/transfers/:id', transfers.getResource)
     router.get('/transfers/:id/state', transfers.getStateResource)
+    router.get('/transfers/:id/hash_preimage', transfers.getPreimage)
```

Comment:
Yes, it returns a hash. The preimage of the hash is
```
{
    id: transfer_id,
    state: state,
    token: sign(sha512(transfer_id + ':' + state))
  }
```
Can you clarify how the hash is used by the sender?